.github/contributing.md

```diff
@@ -50,67 +42,84 @@ To begin:
 
-   ```
-   git checkout -b MY_BRANCH_NAME
+   ```bash
+   git checkout -b <YOUR BRANCH NAME>
    ```
 
 4. Install the dependencies:
-   ```
+   ```bash
    pnpm install
    ```
    If you are on windows, use the `--ignore-scripts` flag
-   ```
+   ```bash
    pnpm install --ignore-scripts
    ```
 
-Now you have got the repo on your local machine, and you're ready to start making your changes!
+Now you have the repo on your local machine, and you're ready to start making your changes!
 
 <br/>
 
 ### Test Your Changes
 
-We use [yalc](https://github.com/wclr/yalc) to test changes locally.
+#### Writing Unit Tests
 
-Install the yalc CLI globally:
+Try to include unit test coverage with your changes where appropriate. We use `vitest` for testing, look for files with the `.test.ts` extension for examples of how to structure your unit tests.
+
+To run your tests, run the following from the package directory (most likely /packages/thirdweb):
 
 ```bash
-pnpm add yalc -g
+pnpm test <YOUR TEST FILE PATH>
 ```
 
-First, create a test project where you can experiment with your changes:
+> Specifying your test file path is optional, but will save time by only running specific tests each time. Before opening your PR, run `pnpm test` from the monorepo root (without specifying your test file) to ensure your changes didn't break any existing tests.
+
+#### Linting
 
-You can create a basic starter project with the `sdk` and `react` packages installed using the CLI:
+We use a linter to maintain best practices across projects. Once your changes are complete (or periodically while making changes), run the linter with the following command from the repo root:
 
 ```bash
-npx thirdweb create --app
+pnpm lint
 ```
 
-Use `yalc` to link your local changes in the monorepo to the test project, by running the following command from your test repo:
+If there are errors, try running `pnpm fix` to auto-fix any basic errors. Other linter errors, like missing documentation, will need to be fixed manually. See existing files for examples of inline documentation.
+
+#### Creating a Local Test Project
+
+You can use [yalc](https://github.com/wclr/yalc) to test changes on a local project.
+
+Install the yalc CLI globally:
 
 ```bash
-yalc add @thirdweb-dev/react # Link the react package
-yalc add @thirdweb-dev/sdk # Link the sdk package
-# etc...
+pnpm add yalc -g
 ```
 
-From the monorepo, run the following command to publish your local changes to the test project:
+First, create a test project where you can experiment with your changes. This can be any node project that uses your changes.
+
+
+Push your local version to your `yalc` store with:
 
 ```bash
 pnpm push
 ```
 
-Now, each time you make a change to the monorepo, you can run `pnpm push` to publish your changes to the test project.
+Link your local changes in the monorepo to the test project, by running the following command from your test repo:
+
+```bash
+yalc add thirdweb
```

Comment:
if there was an underlying dependency change, you might need to install deps on the target project

Reply:
Makes sense, just updated that on line 113. Lmk if that ordering is incorrect